src/dmd/semantic3.d

```diff
@@ -1146,11 +1146,11 @@ private extern(C++) final class Semantic3Visitor : Visitor
 
                             s = s.statementSemantic(sc2);
 
-                            bool isnothrow = f.isnothrow & !(funcdecl.flags & FUNCFLAG.nothrowInprocess);
```

Comment:
I think we were relying on lucky behaviour. Isn't unchecked bitwise `and` (`&`) undefined behaviour?

Reply:
well, it eventually ends up being a 1 or 0 so I doubt that any implementation would do something different, but yeah, better safe than sorry.

Reply:
A `bool` may not always be 0 or 1, in memory, hence my question. I don't know how `&` is defined in D tho, but might produce unintended behaviour, in some cases, especially if the `bool` value comes from a C function.